consul/src/main/scala/io/buoyant/consul/v1/AgentApi.scala

```diff
@@ -0,0 +1,31 @@
+package io.buoyant.consul.v1
+
+import com.twitter.conversions.time._
+import com.twitter.finagle.http
+import com.twitter.finagle.service.Backoff
+import com.twitter.finagle.stats.{DefaultStatsReceiver, StatsReceiver}
+import com.twitter.util.{Closable, Duration, Future}
+
+object AgentApi {
+  def apply(c: Client): AgentApi = new AgentApi(c, s"/$versionString")
+}
+
+class AgentApi(
+  override val client: Client,
+  override val uriPrefix: String,
+  override val backoffs: Stream[Duration] = Backoff.exponentialJittered(1.milliseconds, 5.seconds),
+  override val stats: StatsReceiver = DefaultStatsReceiver
+) extends BaseApi with Closable {
+
+  val agentPrefix = s"$uriPrefix/agent"
+
+  // https://www.consul.io/docs/agent/http/agent.html#agent_self
+  def self(retry: Boolean = false): Future[Self] = {
+    val req = mkreq(http.Method.Get, s"$agentPrefix/self")
+    executeJson[Self](req, retry).map(_.value)
+  }
+}
+
+case class Config(Domain: Option[String])
+
+case class Self(Config: Option[Config])
```

Comment:
please lowercase field names (`domain`, `config`).

please include a ~1-sentence summary comment of each of these types so that a maintainer can get their bearings without having to read consul docs